### ISSUE TITLE
gnome-software-3.24.3 fails to build

### ISSUE THREAD

**Issue by mdorman** [NixOS/nixpkgs#29157]
## Issue description

I attempted to add a gnome desktop to my nixos machine.  I added:

```
xserver = {
  desktopManager.gnome3.enable = true;
  displayManager.gdm.enable = true;
  enable = true;
  libinput.enable = true;
};
```

To `/etc/nixos/configuration.nix`.  This fails to build:

```
builder for ‘/nix/store/3yczvskic4zhvbmzjvvp2sm50inr3hnb-gnome-software-3.24.3.drv’ failed with exit code 2
```

### Steps to reproduce

Simply try to build gnome-software:

```
$ nix-build --no-build-output --attr gnome3.gnome-software /root/nixpkgs
these derivations will be built:
  /nix/store/3yczvskic4zhvbmzjvvp2sm50inr3hnb-gnome-software-3.24.3.drv
building path(s) ‘/nix/store/lr5wn5i39qjxsp37bgkmd9s3j2pxcmch-gnome-software-3.24.3’
builder for ‘/nix/store/3yczvskic4zhvbmzjvvp2sm50inr3hnb-gnome-software-3.24.3.drv’ failed with exit code 2
error: build of ‘/nix/store/3yczvskic4zhvbmzjvvp2sm50inr3hnb-gnome-software-3.24.3.drv’ failed
```

This fails with:

```
  CCLD     gnome-software-cmd
/nix/store/yf4p5w2v4h4i8rja9zw1akp007av624j-binutils-2.28.1/bin/ld: warning: libarchive.so.13, needed by /nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so, not found (try using -rpath or -rpath-link)
/nix/store/yf4p5w2v4h4i8rja9zw1akp007av624j-binutils-2.28.1/bin/ld: warning: libgcab-1.0.so.0, needed by /nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so, not found (try using -rpath or -rpath-link)
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_entry_pathname'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_entry_update_pathname_utf8'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `gcab_cabinet_new@LIBGCAB1_0.0'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_entry_update_hardlink_utf8'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_support_filter_all'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_free'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `gcab_cabinet_extract_simple@LIBGCAB1_0.2'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_error_string'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_entry_hardlink'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_close'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `gcab_file_get_name@LIBGCAB1_0.0'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_open_memory'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_entry_update_symlink_utf8'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_next_header'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_support_format_all'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `gcab_cabinet_load@LIBGCAB1_0.0'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_new'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_entry_symlink'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `gcab_file_set_extract_name@LIBGCAB1_0.2'
/nix/store/vp0gnhi1pckki4w7wfqg7d1g9nm87p5q-appstream-glib-0.7.2/lib/libappstream-glib.so: undefined reference to `archive_read_extract'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:678: gnome-software-cmd] Error 1
make[2]: Leaving directory '/tmp/nix-build-gnome-software-3.24.3.drv-0/gnome-software-3.24.3/lib'
make[1]: *** [Makefile:537: all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-gnome-software-3.24.3.drv-0/gnome-software-3.24.3'
make: *** [Makefile:469: all] Error 2
```

## Technical details

This is a build against nixpkgs@e480b4baeefadeb39c1a9fdcbbebd3808cfe41bc

I have to assume that this is a consequence of b73e3b6095e5d8d952682d4b868e9ac7f633f0c2.  @lheckemann any ideas?

**Comment by mdorman** [comment]
You can also see this in hydra: https://hydra.nixos.org/build/60573669

**Comment by vcunat** [comment]
No, we know why this is already.

**Comment by vcunat** [comment]
See https://github.com/vcunat/nixpkgs/commit/35f89413e288#commitcomment-24154387

**Comment by eqyiel** [comment]
@mdorman a workaround for the time being:

```
environment.gnome3.excludePackages = [
    # gnome-software doesn't build and it wouldn't work with nixos anyway, at
    # least before something like this is done:
    # RFC: Generating AppStream Metadata #15932:
    # https://github.com/NixOS/nixpkgs/issues/15932
    pkgs.gnome3.gnome-software
  ];
```

Perhaps `gnome-software` should be excluded by default, unless someone is motivated to work on https://github.com/NixOS/nixpkgs/issues/15932

What do you think?

cc @jtojnar, @lheckemann, @ryantm, @romildo, @teh, @matthewbauer

**Comment by vcunat** [comment]
No need to.  Shea promised to fix this and did so in 6a17c5a46c9.  It runs now, though it probably isn't able to do anything useful yet :-)

**Comment by vcunat** [comment]
Well, *if* it really isn't able to do anything useful yet, I think it's most likely better to exclude it from the default anyway.
